boltdb.go

```diff
@@ -77,54 +139,76 @@ func (db BoltDB) SaveMessage(mes *Message) error {
 
 // Delete a message from the database based on Id.
 func (db BoltDB) DeleteMessage(mes *Message) error {
+	if mes == nil {
+		return ErrMessageObjectNil
+	}
+
 	err := db.Conn.Update(func(tx *bolt.Tx) error {
-		b, err := tx.CreateBucketIfNotExists([]byte(MESSAGES))
-		if err != nil {
-			return err
-		}
+		b := tx.Bucket([]byte(MESSAGES))
 
 		return b.Delete([]byte(mes.Id))
 	})
 	return err
 }
 
 func (db BoltDB) SaveUser(user *User) error {
+	if user == nil {
+		return ErrUserObjectNil
+	}
+
 	err := db.Conn.Update(func(tx *bolt.Tx) error {
-		b, err := tx.CreateBucketIfNotExists([]byte(USERS))
+		encoded := MarshalUser(user)
+
+		b := tx.Bucket([]byte(USERIDS))
+		err := b.Put([]byte(user.Id), encoded)
 		if err != nil {
 			return err
 		}
 
-		encoded := MarshalUser(user)
+		b = tx.Bucket([]byte(USERNAMES))
+		err = b.Put([]byte(user.Username), encoded)
 		if err != nil {
 			return err
 		}
-		return b.Put([]byte(user.Id), encoded)
+
+		return nil
```

Comment:
Either. It will return the error if there is one, otherwise returns nil. I copied and pasted some code so there are a couple extra lines that aren't needed but don't actually change anything. Good catch.